File: codeforces/cf1407A_ahahahahahahahaha.cpp
// Vicfred
// https://codeforces.com/contest/1407/problem/A
// math, greedy, implementation
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int t;
    cin >> t;

    while(t--) {
        int n;
        cin >> n;

        vector<int> a(n);
        for(auto &item: a)
            cin >> item;

        int zero, one;
        zero = one = 0;

        for(auto item: a) {
            if(item == 0)
                zero += 1;
            else
                one += 1;
        }

        if(one <= n/2) {
            cout << zero << endl;
            for(int i = 0; i < zero; i++)
                cout << 0 << " ";
            cout << endl;
            continue;
        }

        if(one%2 != 0)
            one -= 1;
        cout << one << endl;
        for(int i = 0; i < one; i++)
            cout << 1 << " ";
        cout << endl;
    }

    return 0;
}

File: codeforces/cf1407A_ahahahahahahahaha.py
# uninhm
# https://codeforces.com/contest/1407/problem/A
# greedy

t = int(input())

for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    if a.count(1) <= n//2:
        print(a.count(0))
        print(*[i for i in a if i == 0])
    else:
        if sum(a) % 2 == 0:
            print(a.count(1))
            print(*[i for i in a if i == 1])
        else:
            print(a.count(1)-1)
            print(*([i for i in a if i == 1][:-1]))
